src/lib/lib.rs

```diff
@@ -44,22 +45,28 @@ extern "C" {
     fn log(s: &str);
 }
 
-pub fn exec(src: &str) -> String {
+fn parser_expr(src: &str) -> Expr {
     let mut lexer = Lexer::new(src);
     lexer.lex().unwrap();
     let tokens = lexer.tokens;
+    Parser::new(tokens).parse_all().unwrap()
+}
 
+pub fn forward(engine: &mut Interpreter, src: &str) -> String {
```

Comment:
why `forward` as a name?

Reply:
Would also be worth some comments above this function definition to explain what its purpose is. (compared to exec)

Reply:
Let me see, it was call `run_frame` in RustPython.

I'm not sure what this should be called, `forward` sounds like walk a small step, which means the step by step interpretation.